### PR TITLE
Aglu dev

### PR DESCRIPTION
**Pull request by kvcalvin**
This branch adds documentation for the Agriculture and Land Use component of GCAM.

### CODE REVIEW

File: AgLU.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+### Land Sharing Approach
+
+Economic land use decisions in GCAM are based on a logit model of sharing based on relative inherent profitability of using land for competing purposes. The logit approach was developed my McFadden (1974) to describe the distribution of consumer choice, but a similar mathematical approach works well here for our purposes. The interpretation of this sharing system in GCAM is that there is a distribution of profit behind each competing land use within a region, rather than a single point value. Each competing land use option has a potential average profit over its entire distribution. The share of land allocated to any given use is based on the probability that that use has a highest profit among the competing uses. The relative potential average profits are used in the logit formulation, where an option with a higher average profit will get a higher share than one with a lower average profit. However, at the margin, the profits of competing options are equal. In essence, the sharing can therefore be considered optimal, but the assumption of a distribution of profits makes it very different from a constrained linear optimization approach.
+
```
Comment (bpbond) [inline AgLU.md]: "by McFadden"

File: AgLU.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+## AgLU Inputs and Outputs
+
+This section briefly summarized inputs and outputs, including data sources, spatial and temporal resolution. A more complete description of data is available at: http://www.pnl.gov/main/publications/external/technical_reports/PNNL-21025.pdf.
```
Comment (bpbond) [inline AgLU.md]: "summarizes"

File: AgLU.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+### Vegetation Carbon
+
+For vegetation carbon, we distinguish between two different activities – land expansion and land contraction. In the event that land contracts, i.e., there is less land in the current period than the previous period, we assume that all emissions are released instantaneously.  In this event, the land-use change emissions for the current period from above ground carbon changes are equal to the change in above ground carbon stock. Emissions from above ground carbon changes for all other periods are equal to zero. In the event land expands, we spread the change in carbon stock across time depending on the length of time it takes for the vegetation to mature.  For crops, the mature age is typically set a 1 year, meaning that all carbon uptake occurs instantaneously. For forests, however, we assume it takes anywhere from 30 years to 100 years to uptake all of the carbon. We use an example of the Bertalanffy-Richards function.
+
+### Soil Carbon
+
+For soil carbon, we assume emissions and uptake is exponential, where the half life depends on the region. Colder regions have longer half-lives.
```
Comment (bpbond) [inline AgLU.md]: For clarity, perhaps "we assume both emissions and uptake are exponential, where the exponential half-life depends"

File: AgLU.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+### Carbon Stocks
+
+GCAM tracks carbon stocks by calculating and storing cumulative land-use change emissions. As land expands, we add future uptake to the carbon stock and as land contracts we subtract future emissions from the carbon stock.
```
Comment (bpbond) [inline AgLU.md]: Maybe "GCAM tracks carbon stocks by calculating and storing cumulative land-use change emissions, and then applying those emissions as time proceeds. As land expands, we compute future uptake to be added to the carbon stock, and as land contracts we compute future emissions to subtract from the carbon stock."  ?

**Comment by rplzzz** [comment]
The terms in the two description lists weren't really being emphasized enough, so I tried a couple of ways of making them a little more visible.  In the Input section I used the html description list, along with a few tweaks to the style sheet.  In the Output section I left the list in ordinary paragraph style and just changed the italicized terms to bold.

If everyone could take a quick look at it and let me know which version they like better, I'll update them both to that style.

**Comment by rplzzz** [comment]
Also, I changed the mixed-case `AgLU.md` to lower case.  There is enough inconsistency in the way different systems treat case differences in file names that I think it's best to just stick to single case.

File: aglu.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+With the specification above, we can make substitution across categories more or less difficult by choosing lower or higher logit parameters. We can also effectively combine nests by making the logit exponents the same. For example, if we assign the same logit exponent for each level, this is equivalent to putting all uses under one nest.  One implication of having equal exponents is that it would not matter in which node we placed new crop options like biomass.
```
Comment (rplzzz) [inline aglu.md]: @kvcalvin, is it true that making the logit exponents the same is equivalent to putting them all in a single nest?  The math behind that is not obvious to me.

A quick test shows that it's _not_ true using the weighted average formula for the cost of a nest (which is what we use in the energy system).  It also doesn't seem to be true when you use the average cost formula from Clarke and Edmonds, though I concede that in coding that one from memory I may have messed it up subtly.  Here are my numbers with a logit exponent of 3 and p1=1, p2=3, and p3=2 (in the nested version 1 and 2 are in one nest, with 3 in a nest by itself):

```
python share.py
nested shares    1: 0.0270868606265  2: 0.731345236916  3: 0.241567902457
alt nested       1: 0.0273285414795  2: 0.737870619946  3: 0.234800838574
unnested shares  1: 0.0277777777778  2: 0.75  3: 0.222222222222
```

Anyhow, I think we either need to support this claim or remove it.

**Comment by kvcalvin** [comment]
@rplzzz: For AgLU, we use the original Clarke & Edmonds formula. I haven't done the math in a few years, but think that was true before. I can check again.

**Comment by rplzzz** [comment]
Ok, here are the changes I made (some of these were in earlier commits that we have already discussed):
- Rename AgLU.md to aglu.md
- Remove references to GCAM 4.2:  The logic here is that generally we should avoid referring to the current version number, so that when we update to a new version we don't have to go back and change the version number for everything that hasn't changed.
- Minor word usage tweaks.
- Description list formatting.
- Change figure numbering -- there didn't seem to be a figure 1.  Also, add the .fig class to the figure paragraph.  When this gets merged with the logit branch, that will cause the figure to be centered and the caption to appear in a distinctive style.
- Add cite to David Ricardo's book. Since we mentioned him in the text, we might as well give him a hand with his H-index.

**Comment by rplzzz** [comment]
Unless anyone else wants to weigh in, I think this one is all set to merge.

:shipit:

**Pull request merged**